File: mobile/src/interfaces/currencies.ts
// Copyright (c) The Diem Core Contributors
// SPDX-License-Identifier: Apache-2.0

export type DiemCurrency = "XDM" | "Coin1" | "Coin2";

export type FiatCurrency = "USD" | "EUR" | "GBP" | "CHF" | "CAD" | "AUD" | "NZD" | "JPY";

export type DiemCurrenciesSettings = {
  [key in DiemCurrency]: {
    name: string;
    symbol: DiemCurrency;
    sign: string;
  };
};

export type FiatCurrenciesSettings = {
  [key in FiatCurrency]: {
    symbol: FiatCurrency;
    sign: string;
  };
};

export type Rates = {
  [key in DiemCurrency]: {
    [key in FiatCurrency | DiemCurrency]: number;
  };
};

File: liquidity-old/diem_utils/types/currencies.py
# Copyright (c) The Diem Core Contributors
# SPDX-License-Identifier: Apache-2.0

import typing
from enum import Enum


class DiemCurrency(str, Enum):
    Coin1 = "Coin1"


class FiatCurrency(str, Enum):
    USD = "USD"
    EUR = "EUR"
    GBP = "GBP"
    CHF = "CHF"
    CAD = "CAD"
    AUD = "AUD"
    NZD = "NZD"
    JPY = "JPY"


Currencies = typing.Union[FiatCurrency, DiemCurrency]

DEFAULT_DIEM_CURRENCY = DiemCurrency.Coin1.value
